### DISCUSSION TITLE
Setting up xDrip as my Uploader with Google Cloud

### DISCUSSION THREAD

**Discussion by mcrews1** [NightscoutFoundation/xDrip#2723] [Q&A]
When I scan the QR code to configure Nightscount in xDrip, I get the error failed. I am building using a browser and trying to eliminate Heroku. I tried entering in the URL but that did not work either.

**Comment by Navid200** [comment]
You opened another thread asking for support.
I responded.  
Are you going to respond to that?  Is that problem resolved?  if yes, you should mention that in the other thread.
Others find these threads who may have similar problems.

**Reply by mcrews1** [reply]
I thought I edited it to say Resolved.  Let me check to ensure I saved it. My apologies. That issue is fixed.

**Reply by mcrews1** [reply]
@Navid200 The above error is not fixed however!!

**Reply by Navid200** [reply]
Post a screenshot of the status page.

**Reply by mcrews1** [reply]
![](https://user-images.githubusercontent.com/37308938/223606744-90ffb83c-94ae-46da-9f11-7d66e075676b.png)

**Comment by Navid200** [comment]
Can you post a screenshot of the status page?

**Reply by mcrews1** [reply]
![image](https://user-images.githubusercontent.com/37308938/223606744-90ffb83c-94ae-46da-9f11-7d66e075676b.png)

Error says:
Cannot GET / API /vi

**Reply by Navid200** [reply]
So, when you use SSH to connect, you don't get a menu?

**Reply by mcrews1** [reply]
Yes. I clicked on xDrip setup and it gave me the QR code.

**Reply by Navid200** [reply]
Did you install Nightscout on Google Cloud following this?  https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html

**Reply by mcrews1** [reply]
Yes...I followed every step (I thought)...read and then watched the videos but I have no idea what to do when I get to the xDrip QR code!

**Reply by mcrews1** [reply]
Should I delete and start over?

**Reply by Navid200** [reply]
I have asked you to post a screenshot of the status page several times.  I am trying to understand what the problem is.  But, when you post cannot GET /api/vi
instead of posting a screenshot of the status page, I have to wonder if we understand each other.

**Reply by Navid200** [reply]
> Should I delete and start over?

I have no idea if you don't let me see the status page.

**Reply by mcrews1** [reply]
I have not gotten any errors. Where can I show you the status? I am not a programmer so this is so complicated for me. Where do I enter my private URL 
![image](https://user-images.githubusercontent.com/37308938/223610509-30c3917d-1ba1-4017-9039-9c6d3ee58ada.png)

And my apologies..I am just totally lost!

**Reply by Navid200** [reply]
This explains how to get to the status page:
https://navid200.github.io/xDrip/docs/Nightscout/Status.html

**Reply by mcrews1** [reply]
![image](https://user-images.githubusercontent.com/37308938/223611438-4abbf29d-d761-4c16-9b25-5bf2de99e49a.png)

**Reply by Navid200** [reply]
Great

When you enter the hostname shown on that same page under "Hostname and password", into a web site, do you see your Nightscout?
Did it ask you to authenticate?
Did you?

**Reply by mcrews1** [reply]
I am assuming that is the https://apisecret@dnssubdomain/api/vi ?

**Reply by mcrews1** [reply]
If I copy and paste that from the xDrip setup script into the address bar of a web browser, I get the website cannot be found
ERR_FAILED

**Reply by Navid200** [reply]
The hostname is shown on the status page when you click at the bottom on Hostname and password.
It has no /api/vi at the end and it has no @ in it.

**Reply by mcrews1** [reply]
Yes I have Nightscout but there is no data?

**Reply by Navid200** [reply]
The menu showed the QR code, and you scanned it with xDrip?

**Comment by mcrews1** [comment]
I do not understand how to scan with xDrip? Do I need to download something on my phone? I get the "Cannot GET /api/v1/" error when I scan. This is my question up at the top...I do not know where and how to scan for xdrip?

**Reply by Navid200** [reply]
In xDrip, you go to Settings &#8722;> Auto Configure

That makes xDrip start scanning.

**Comment by mcrews1** [comment]
I have never used xDrip so not sure it is loaded anywhere. I do not know how to get to xDrip...

**Reply by Navid200** [reply]
What app do you use as your CGM?

**Reply by mcrews1** [reply]
Dexcom

**Reply by Navid200** [reply]
This explains how you can use xDrip instead: https://navid200.github.io/xDrip/docs/Starting-G6.html

There are people who don't use xDrip and still use Nightscout.

This is a group that helps with Nightscout:  https://www.facebook.com/groups/cgminthecloud
You can post there and ask how you can upload to Nightscout from Dexcom app.

This has nothing to do with Nightscout on Google Cloud.  Youc an choose to use different platforms for hosting your Nightscout.  After you have accomplished that, you will need to set up your uploader.  That is going to be the same regardless of which platform you use.

**Reply by mcrews1** [reply]
I have never used xDrip and this is where my issue is...I cannot figure out where or how to download. I am ready to give up and use something else. This cannot be so complicated. When I go to git.hub and click on the latest release, it goes to downloads on my phone and shows as blank. This process is so frustrating...from a non-programmer point of view.

**Reply by Navid200** [reply]
I'm sorry about this.  I will try again to explain:

xDrip is an app that can do a lot.
You are in charge of controlling your diabetes.  We will never tell you what to do.

If you have never used Nightscout before, please go to this page and ask questions for help:
https://www.facebook.com/groups/cgminthecloud

xDrip is only one of many apps one can use as a continuous glucose monitor (CGM).  
If you decide to use xDrip as your CGM, you can.  You will then be able to easily set it up to upload to Nightscout.

But, if you set up Nightscout and then think that you can use xDrip to upload readings to Nightscout, you have misunderstood.  That is not the right approach.

Your questions suggest that you already have a CGM you are happy with and for some reason you have decided to set up Nightscout.  That's all fine.  But, you need to figure out how your CGM needs to be configured as well as how Nightscout needs to be configured.
The place to go and look for answers about those is here: https://www.facebook.com/groups/cgminthecloud

xDrip is not an app you install in order to upload readings from your existing CGM to Nightscout.

You can pay a monthly fee and Nightscout will be set up for you.  Again, go to that site and ask and they will give you all options available for setting up Nightscout:  https://www.facebook.com/groups/cgminthecloud

**Reply by mcrews1** [reply]
Now I am getting a bad gateway error...

**Reply by Navid200** [reply]
What did you do since the time it was working fine?

**Reply by mcrews1** [reply]
Resolved

**Reply by mcrews1** [reply]
Who knows what I did! I tried 100 different things and it is working...and also updating which it was not doing before.
